docs/pipeline-resources.md

```diff
@@ -0,0 +1,83 @@
+### Pipeline Resources
+
+## Git Resource
```

Comment:
so cool, thanks for adding docs!

it would be good at add a link from https://github.com/knative/build-pipeline#resources to here as well, since folks won't know this doc exists